packages/client/mutations/CreatePollMutation.ts

```diff
@@ -20,21 +31,60 @@ const mutation = graphql`
           message
         }
       }
-      ...CreatePollMutation_poll @relay(mask: false)
+      ...CreatePollMutation_meeting @relay(mask: false)
     }
   }
 `
 
-const CreatePollMutation: StandardMutation<TCreatePollMutation> = (
+export const createPollMeetingUpdater: SharedUpdater<CreatePollMutation_meeting> = (
+  payload,
+  {store}
+) => {
+  const poll = payload.getLinkedRecord('poll')
+  if (!poll) {
+    console.warn('No poll object found, skipping!')
+    return
+  }
+  const discussionId = poll.getValue('discussionId')
+  if (discussionId) {
+    const threadConn = getDiscussionThreadConn(store, discussionId)
+    safePutNodeInConn(threadConn, poll, store, 'threadSortOrder', true)
+  }
+}
+
+interface Handlers extends LocalHandlers {
+  localPoll: Poll_poll
+}
+
+const CreatePollMutation: StandardMutation<TCreatePollMutation, Handlers> = (
   atmosphere,
   variables,
-  {onError, onCompleted}
+  {localPoll, onError, onCompleted}
 ) => {
   return commitMutation<TCreatePollMutation>(atmosphere, {
     mutation,
     variables,
-    optimisticUpdater: (store) => {
-      const {} = variables
```

Comment:
As poll is already in the relay store might not need an optimistic updater. The only problem with that solution is, after clicking 'Start poll' the poll is still in the 'creating' state. We might want to extend the gql `Poll` type on the client and add a property `isLocalPoll` or something like that and update it to `false` in optimistic updater. This would make the `Poll` look like an already created poll just after clicking the `Start poll` button. Let me know what do you think! 💡